### PR TITLE
Replace error status with exceptions

### PR DESCRIPTION
**Pull request by andres-sommariva**
cc @sebastian-scialabba , @ezequielmatalon

### CODE REVIEW

File: pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java [hunk 1]
```diff
@@ -48,10 +50,15 @@
 
 
 	@Override
-	public CreateTxResponse createTransaction(CreateTxRequest request) {
+	public CreateTxResponse createTransaction(CreateTxRequest request) throws TransactionFailedException {
 		ProviderCredentials credentials = providerConfig.getProviderCredentials(ServiceConfigType.PX_FUSION);
 		ProviderSettings settings = providerConfig.getProviderSettings(ServiceConfigType.PX_FUSION);
 		CreateTxResponse response = pxFusionClient.getTransactionId(request, settings, credentials);
+		
+		if (!Status.SUCCESS.equals(response.getStatus())) {
```
Comment (sebastian-scialabba) [inline pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java]: You could write `response.getStatus() != Status.SUCCESS`

File: pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java [hunk 2]
```diff
@@ -64,39 +71,62 @@ public GetTxStatusResponse getTransactionStatus(GetTxStatusRequest request) {
...
 	}
 
 	@Override
-	public ValidateCCResponse validateCreditCard(ValidateCCRequest request) {
+	public ValidateCCResponse validateCreditCard(ValidateCCRequest request) throws TransactionFailedException {
 		try {
 			CreditCardValidator.validate(request.getCreditCard());
 		} catch (InvalidCreditCardException e) {
-			return ValidateCCResponse.builder().status(Status.FAILED).statusMessage(e.getMessage()).build();
+			throw new TransactionFailedException(e.getMessage());
```
Comment (ezequiel-matalon) [inline pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java]: Chain exceptions

File: pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java [hunk 3]
```diff
@@ -64,39 +71,62 @@ public GetTxStatusResponse getTransactionStatus(GetTxStatusRequest request) {
 	}
 
 	@Override
-	public CancelTxResponse cancelTransaction(CancelTxRequest request) {
+	public CancelTxResponse cancelTransaction(CancelTxRequest request) throws TransactionFailedException {
 		ProviderCredentials credentials = providerConfig.getProviderCredentials(ServiceConfigType.PX_FUSION);
 		ProviderSettings settings = providerConfig.getProviderSettings(ServiceConfigType.PX_FUSION);
 		CancelTxResponse response = pxFusionClient.cancelTransaction(request, settings, credentials);
+		
+		if (!Status.SUCCESS.equals(response.getStatus())) {
```
Comment (sebastian-scialabba) [inline pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java]: You could write `response.getStatus() != Status.SUCCESS`

File: pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java [hunk 4]
```diff
@@ -64,39 +71,62 @@ public GetTxStatusResponse getTransactionStatus(GetTxStatusRequest request) {
...
 	}
 
 	@Override
-	public RefundTxResponse refundTransaction(RefundTxRequest request) {
+	public RefundTxResponse refundTransaction(RefundTxRequest request) throws TransactionFailedException, TransactionDeclinedException {
 		ProviderCredentials credentials = providerConfig.getProviderCredentials(ServiceConfigType.PX_WEB_SERVICE);
 		ProviderSettings settings = providerConfig.getProviderSettings(ServiceConfigType.PX_WEB_SERVICE);
 		RefundTxResponse response = pxWebServiceClient.refundTransaction(request, settings, credentials);
+		
+		if (Status.FAILED.equals(response.getStatus())) {
```
Comment (sebastian-scialabba) [inline pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java]: You could write `response.getStatus() == Status.FAILED`

File: pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java [hunk 5]
```diff
@@ -64,39 +71,62 @@ public GetTxStatusResponse getTransactionStatus(GetTxStatusRequest request) {
...
+			throw new TransactionFailedException();
+		} else if (Status.DECLINED.equals(response.getStatus())) {
```
Comment (sebastian-scialabba) [inline pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java]: You could write `response.getStatus() == Status.DECLINED`

File: pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java [hunk 6]
```diff
@@ -64,39 +71,62 @@ public GetTxStatusResponse getTransactionStatus(GetTxStatusRequest request) {
...
 		} catch (InvalidCreditCardException e) {
-			return ValidateCCResponse.builder().status(Status.FAILED).statusMessage(e.getMessage()).build();
+			throw new TransactionFailedException(e.getMessage());
 		}
+		
 		ProviderCredentials credentials = providerConfig.getProviderCredentials(ServiceConfigType.PX_WEB_SERVICE);
 		ProviderSettings settings = providerConfig.getProviderSettings(ServiceConfigType.PX_WEB_SERVICE);
 		ValidateCCResponse response = pxWebServiceClient.validateCreditCard(request, settings, credentials);
+		
+		if (!Status.APPROVED.equals(response.getStatus())) {
```
Comment (sebastian-scialabba) [inline pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java]: You could write `response.getStatus() != Status.APPROVED`

File: pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java [hunk 7]
```diff
@@ -64,39 +71,62 @@ public GetTxStatusResponse getTransactionStatus(GetTxStatusRequest request) {
...
 	}
 
 	@Override
-	public UpdateCCResponse updateCreditCard(UpdateCCRequest request) {
+	public UpdateCCResponse updateCreditCard(UpdateCCRequest request) throws TransactionFailedException {
 		ProviderCredentials credentials = providerConfig.getProviderCredentials(ServiceConfigType.PX_WEB_SERVICE);
 		ProviderSettings settings = providerConfig.getProviderSettings(ServiceConfigType.PX_WEB_SERVICE);
 		UpdateCCResponse response = pxWebServiceClient.updateCreditCard(request, settings, credentials);
+		
+		if (!Status.SUCCESS.equals(response.getStatus())) {
```
Comment (sebastian-scialabba) [inline pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java]: You could write `response.getStatus() != Status.SUCCESS`

File: pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java (hunk 1 above)
Comment (ezequiel-matalon) [inline pg-api-parent/pg-api-px-fusion-impl/src/main/java/com/appdirect/pg/api/px/service/PXPaymentGatewayProviderImpl.java]: Yoda condition, so much better than the other way, prevents NPE

https://en.wikipedia.org/wiki/Yoda_conditions

**Comment by andres-sommariva** [comment]
@ezequiel-matalon, @sebastian-scialabba hice los cambios mencionados!

**Pull request merged**